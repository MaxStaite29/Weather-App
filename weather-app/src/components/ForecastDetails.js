import React from "react";

function ForecastDetails({ forecast }) {
  const formattedDate = new Date(forecast.date).toDateString();
  return (
    <div className="forecast-details" data-testid="forecast-details">
      <div className="forecast-details__date">{formattedDate}</div>
      <div className="forecast-details__min-temp">
        Min Temperature: {forecast.temperature.min}&deg;C
      </div>
      <div className="forecast-details__max-temp">
        Max Temperature: {forecast.temperature.max}&deg;C
      </div>
      <div className="forecast-details__humidity">
        Humidity: {forecast.humidity}%
      </div>
      <div className="forecast-details__wind-speed">
        Wind Speed: {forecast.wind.speed} mph
      </div>
      <div className="forecast-details__wind-direction">
        Wind Direction: {forecast.wind.direction}
      </div>
    </div>
  );
}

export default ForecastDetails;
