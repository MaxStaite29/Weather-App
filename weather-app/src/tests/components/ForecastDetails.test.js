import React from "react";
import { render } from "@testing-library/react";
import ForecastDetails from "../../components/ForecastDetails";

describe("ForecastDetails", () => {
  const validProps = {
    date: 1525046400000,
    temperature: {
      min: 12,
      max: 22,
    },
    humidity: 80,
    wind: {
      speed: 150,
      direction: "e",
    },
  };

  it("renders correctly", () => {
    const { asFragment } = render(<ForecastDetails forecast={validProps} />);

    expect(asFragment()).toMatchSnapshot();
  });

  it("renders correct values for props", () => {
    const { getByText } = render(<ForecastDetails forecast={validProps} />);

    expect(getByText(/Mon Apr 30 2018/i)).toHaveClass("forecast-details__date");
    expect(getByText(/22°C/i)).toHaveClass("forecast-details__max-temp");
    expect(getByText(/12°C/i)).toHaveClass("forecast-details__min-temp");
    expect(getByText(/80%/i)).toHaveClass("forecast-details__humidity");
    expect(getByText(/150 mph/i)).toHaveClass("forecast-details__wind-speed");
    expect(getByText(/Direction: e/i)).toHaveClass(
      "forecast-details__wind-direction",
    );
  });
});
