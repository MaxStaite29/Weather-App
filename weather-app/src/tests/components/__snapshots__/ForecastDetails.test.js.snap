// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastDetails renders correctly 1`] = `
<DocumentFragment>
  <div
    class="forecast-details"
    data-testid="forecast-details"
  >
    <div
      class="forecast-details__date"
    >
      Mon Apr 30 2018
    </div>
    <div
      class="forecast-details__min-temp"
    >
      Min Temperature: 12°C
    </div>
    <div
      class="forecast-details__max-temp"
    >
      Max Temperature: 22°C
    </div>
    <div
      class="forecast-details__humidity"
    >
      Humidity: 80%
    </div>
    <div
      class="forecast-details__wind-speed"
    >
      Wind Speed: 150 mph
    </div>
    <div
      class="forecast-details__wind-direction"
    >
      Wind Direction: e
    </div>
  </div>
</DocumentFragment>
`;
